src/uct/ib/base/ib_device.c

```diff
@@ -386,6 +389,26 @@ ucs_status_t uct_ib_device_port_check(uct_ib_device_t *dev, uint8_t port_num,
         return UCS_ERR_UNSUPPORTED;
     }
 
+    if (md->subnet_filter) {
+        int match = 0;
+
+        for (i = 0; i < uct_ib_device_port_attr(dev, port_num)->gid_tbl_len; i++) {
```

Comment:
i think would be better to compare only the GID which is provided by UCX_IB_GID_INDEX
so UCX_IB_GID_INDEX variable need to be moved from uct_ib_iface to uct_ib_md